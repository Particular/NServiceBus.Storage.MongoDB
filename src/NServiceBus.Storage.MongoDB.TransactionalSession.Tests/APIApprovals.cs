namespace NServiceBus.Storage.MongoDB.Tests;

using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;
using TransactionalSession;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void Approve()
    {
        var publicApi = typeof(MongoOpenSessionOptions).Assembly.GeneratePublicApi(new ApiGeneratorOptions
        {
            ExcludeAttributes =
            [
                "System.Runtime.Versioning.TargetFrameworkAttribute",
                "System.Reflection.AssemblyMetadataAttribute"
            ]
        });
        Approver.Verify(publicApi);
    }
}