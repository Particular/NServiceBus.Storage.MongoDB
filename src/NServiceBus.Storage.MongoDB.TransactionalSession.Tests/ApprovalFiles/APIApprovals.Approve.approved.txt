namespace NServiceBus.TransactionalSession
{
    public sealed class MongoSessionOptions : NServiceBus.TransactionalSession.OpenSessionOptions
    {
        public MongoSessionOptions() { }
    }
    public static class MongoTransactionalSessionExtensions
    {
        public static NServiceBus.PersistenceExtensions<NServiceBus.MongoPersistence> EnableTransactionalSession(this NServiceBus.PersistenceExtensions<NServiceBus.MongoPersistence> persistenceExtensions) { }
    }
}